### PR TITLE
Backend/feat/bus service api temp

### PR DESCRIPTION
**Pull request by Radhika1708**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: .DS_Store
Comment (khuzema786) [inline .DS_Store:1]: Revert this file

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 1]
```diff
@@ -488,6 +503,10 @@ FRFSConfig:
     freeTicketInterval: "9999"
     maxFreeTicketCashback: "0"
     busStationTtl: "1800"
+    straightLineDistance: "4"
+    radius: "1"
+    straightLineDistance: "4"
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml]: Repeated twice. Can be reverted.
Also Default values keep as below:
straightLineDistance: 5000
radius: 3000

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 2]
```diff
@@ -552,6 +571,7 @@ RouteStopMapping:
     routeCode: Text
     stopCode: Text
     stopName: Text
+    routeName: Text
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:572]: Not Required, Can be reverted.

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 3]
```diff
@@ -569,13 +589,28 @@ RouteStopMapping:
...
+  fromTType:
+    stopPoint: Kernel.External.Maps.Types.LatLong lat lon|E
   constraints:
     routeCode: PrimaryKey
     stopCode: PrimaryKey
   queries:
     findByRouteCode:
       kvFunction: findAllWithKV
       where: routeCode
+    findByStopCode:
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:609]: Not Required, Can be reverted.

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/Merchant.yaml [hunk 4]
```diff
@@ -276,8 +276,10 @@ MerchantServiceUsageConfig:
     notifyPerson: NotificationService
     getDistances: MapsService
     getRoutes: MapsService
+    getFrfsAutocompleteDistances: MapsService
     snapToRoad: MapsService
     getPlaceName: MapsService
+    getFrfsAutocompleteDistances: MapsService
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/Merchant.yaml:282]: Repeated Fields. Please Revert One of them.

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 5]
```diff
@@ -469,6 +482,8 @@ FRFSConfig:
     customDates: '[Text]'
     customEndTime: Text
     cancellationReasonId: Maybe Text
+    straightLineDistance: Maybe Meters
+    radiusVal: Maybe Meters
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:486]: Correct the naming: `radiusVal -> radius`

File: Backend/dev/migrations-read-only/rider-app/frfs_config.sql
Comment (khuzema786) [inline Backend/dev/migrations-read-only/rider-app/frfs_config.sql:1]: Revert this file and regenerate. 
**_Do not delete existing files and regenerate:_** As based on the Query history there are checks added to DSL to prevent usage of Queries that has caused issues in Production due to Scale, Backward Compatibility etc.

File: Backend/dev/migrations-read-only/rider-app/merchant_service_usage_config.sql
Comment (khuzema786) [inline Backend/dev/migrations-read-only/rider-app/merchant_service_usage_config.sql:1]: Revert this file and regenerate.
Do not delete existing files and regenerate: As based on the Query history there are checks added to DSL to prevent usage of Queries that has caused issues in Production due to Scale, Backward Compatibility etc.

File: Backend/dev/migrations-read-only/rider-app/route_stop_mapping.sql [hunk 6]
```diff
@@ -3,11 +3,12 @@ CREATE TABLE atlas_app.route_stop_mapping ();
 ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN merchant_id character varying(36) NOT NULL;
 ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN merchant_operating_city_id character varying(36) NOT NULL;
 ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN route_code text NOT NULL;
+ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN route_name text NOT NULL;
 ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN sequence_num integer NOT NULL;
 ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN stop_code text NOT NULL;
 ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN stop_name text NOT NULL;
-ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN stop_lat double precision NOT NULL;
-ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN stop_lon double precision NOT NULL;
+ALTER TABLE atlas_app.route_stop_mapping ADD COLUMN lat double precision NOT NULL;
```
Comment (khuzema786) [inline Backend/dev/migrations-read-only/rider-app/route_stop_mapping.sql:10]: Revert this change

File: Backend/dev/migrations-read-only/rider-app/station.sql [hunk 7]
```diff
@@ -5,20 +5,35 @@ ALTER TABLE atlas_app.station ADD COLUMN code text NOT NULL;
...
 
-ALTER TABLE atlas_app.station ALTER COLUMN merchant_operating_city_id SET NOT NULL;
+ALTER TABLE atlas_app.station ADD COLUMN time_bounds text  default 'Unbounded';
 
 
 ------- SQL updates -------
 
-ALTER TABLE atlas_app.station ALTER COLUMN merchant_id SET NOT NULL;
\ No newline at end of file
+ALTER TABLE atlas_app.station ALTER COLUMN time_bounds SET NOT NULL;
```
Comment (khuzema786) [inline Backend/dev/migrations-read-only/rider-app/station.sql:24]: Revert this Change

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 8]
```diff
@@ -469,6 +482,8 @@ FRFSConfig:
     customDates: '[Text]'
     customEndTime: Text
     cancellationReasonId: Maybe Text
+    straightLineDistance: Maybe Meters
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:485]: Make Domain Non Maybe and Beam type as Maybe.

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml (hunk 5 above)
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:486]: Make Domain Non Maybe and Beam type as Maybe.

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs [hunk 9]
```diff
@@ -629,3 +634,154 @@ getFrfsConfig (pId, mId) opCity = do
...
+
+      currentTime <- getCurrentTime
+
+      let serviceableStops = DTB.findBoundedDomain allStops currentTime
+
+      let filteredStops =
+            filter
+              ( \stop ->
+                  let straightLineDist = highPrecMetersToMeters (CD.distanceBetweenInMeters origin (LatLong (fromMaybe merchantOpCity.lat stop.lat) (fromMaybe merchantOpCity.long stop.lon)))
+                   in straightLineDist <= fromMaybe (Meters (maxBound :: Int)) frfsConfig.straightLineDistance
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs:675]: If Domain type of `frfsConfig.straightLineDistance` becomes Non Maybe then fromMaybe can be removed.

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs [hunk 10]
```diff
@@ -629,3 +634,154 @@ getFrfsConfig (pId, mId) opCity = do
...
+              { origins = NonEmpty.fromList [origin],
+                destinations = NonEmpty.fromList transformedStops,
+                distanceUnit = Meter,
+                sourceDestinationMapping = Just Maps.ManyToMany,
+                travelMode = Just CAR
+              }
+
+      stopsDistanceResp <- Maps.getFrfsAutocompleteDistances mId merchantOpCity.id stopsReq
+
+      let finalStops = filter (\stopResp -> stopResp.distance <= fromMaybe (Meters (maxBound :: Int)) (frfsConfig.radiusVal)) (NonEmpty.toList stopsDistanceResp)
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/FRFSTicketService.hs:703]: If Domain type of `frfsConfig.radius` becomes Non Maybe then fromMaybe can be removed.

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml (hunk 1 above)
Comment (Radhika1708) [reply Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml]: Yes, sorry bhaiya, I forgot, that these are of Meters type

**Review by khuzema786 — APPROVED** [review]
